src/simulation/TxGenerator.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+
+#include "main/Application.h"
+#include "test/TestAccount.h"
+#include "test/TxTests.h"
+
+namespace medida
+{
+class Counter;
+}
+namespace stellar
+{
+// Config settings for SOROBAN_CREATE_UPGRADE
+struct SorobanUpgradeConfig
+{
+    // Network Upgrade Parameters
+    uint32_t maxContractSizeBytes{};
+    uint32_t maxContractDataKeySizeBytes{};
+    uint32_t maxContractDataEntrySizeBytes{};
+
+    // Compute settings for contracts (instructions and memory).
+    int64_t ledgerMaxInstructions{};
+    int64_t txMaxInstructions{};
+    uint32_t txMemoryLimit{};
+
+    // Ledger access settings for contracts.
+    uint32_t ledgerMaxReadLedgerEntries{};
+    uint32_t ledgerMaxReadBytes{};
+    uint32_t ledgerMaxWriteLedgerEntries{};
+    uint32_t ledgerMaxWriteBytes{};
+    uint32_t ledgerMaxTxCount{};
+    uint32_t txMaxReadLedgerEntries{};
+    uint32_t txMaxReadBytes{};
+    uint32_t txMaxWriteLedgerEntries{};
+    uint32_t txMaxWriteBytes{};
+
+    // Contract events settings.
+    uint32_t txMaxContractEventsSizeBytes{};
+
+    // Bandwidth related data settings for contracts
+    uint32_t ledgerMaxTransactionsSizeBytes{};
+    uint32_t txMaxSizeBytes{};
+
+    // State Archival Settings
+    uint32_t bucketListSizeWindowSampleSize{};
+    uint32_t evictionScanSize{};
+    uint32_t startingEvictionScanLevel{};
+};
+
+class TxGenerator
+{
+  public:
+    struct ContractInstance
+    {
+        // [wasm, instance]
+        xdr::xvector<LedgerKey> readOnlyKeys;
+        SCAddress contractID;
+    };
+
+    using TestAccountPtr = std::shared_ptr<TestAccount>;
+    TxGenerator(Application& app);
+
+    void createRootAccount();
+    void updateMinBalance();
+
+    bool loadAccount(TestAccount& account, Application& app);
```

Comment:
nit: Does this really need to take app as an argument?  We already have `mApp`, right?

Reply:
Good point. Fixed.